### PR TITLE
Improved quality of LTT unit tests and some code structure

### PR DESCRIPTION
**Pull request by evanchia-ly-sdets**
Signed-off-by: evanchia-ly-sdets <evanchia@amazon.com>

## What does this PR do?
Removes unused/deprecated code from LTT. Improves quality of unit tests. Minor improvements to rerun command formatting.

## How was this PR tested?
Locally ran smoke and main test suite.

### CODE REVIEW

**Review by AMZN-scspaldi — APPROVED** [review]

File: Tools/LyTestTools/tests/unit/test_artifact_manager.py [hunk 1]
```diff
@@ -24,60 +24,66 @@
 ROOT_LOG_FOLDER = os.path.join("TestResults", DATE, "pytest_results")
 
 
-@mock.patch('os.makedirs', mock.MagicMock())
```
Comment (Kadino) [inline Tools/LyTestTools/tests/unit/test_artifact_manager.py:27]: I'm okay with this change, however the benefit of mocking at this class/suite level is that it assures that directory creation never gets accidentally added back in during test execution.

File: Tools/LyTestTools/tests/unit/test_artifact_manager.py [hunk 2]
```diff
@@ -24,60 +24,66 @@
...
+        mock_get_collision.assert_called_once_with(os.path.join(self.mock_root, test_name), create_amount)
 
     @mock.patch('os.path.exists', mock.MagicMock(return_value=True))
-    def test_SetTestNameAmount_ValidNamePathExists_NoPathCreatedSetsAttributes(self):
+    def test_SetDestPathAmount_ValidFilePathExists_NoFilePathCreatedSetsAttributes(self):
         test_name = 'dummy'
-        create_amount = 5  # Ignored because the path already exists.
-        updated_path = "{}".format(os.path.join(self.mock_artifact_manager.artifact_path, test_name),
-                                   create_amount)
+        updated_path = os.path.join(self.mock_artifact_manager.artifact_path, test_name)
```
Comment (Kadino) [inline Tools/LyTestTools/tests/unit/test_artifact_manager.py:74]: minor: this value isn't updated/modified, so I would name it expected_path

**Review by Kadino — COMMENTED** [review]
I approve most of these changes, but one part looks suspect:

File: Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/terminal_report.py [hunk 3]
```diff
@@ -11,39 +11,39 @@
...
+        terminalreporter.currentfspath = 1
+        terminalreporter.ensure_newline()
     else:
-        terminalreporter.write_line("Error, Test node id list is empty!")
+        terminalreporter.write_line("Error: Cannot provide rerun commands because test node id list is empty!")
 
 
-def pytest_terminal_summary(terminalreporter, exitstatus, config):
+def pytest_terminal_summary(terminalreporter, config):
     """
```
[the remark this answers is not in the document]
Comment (evanchia-ly-sdets) [reply Tools/LyTestTools/ly_test_tools/_internal/pytest_plugin/terminal_report.py:37]: It seems to function still and yes it's being executed, but I'll add it back as I'd also expect this to be required as a pytest hook. I'm actually surprised that it works still.

**Review by Kadino — APPROVED** [review]

**Pull request merged**